### PR TITLE
[Chore] Unit test on SurveyRepo > getSurveys

### PR DESCRIPTION
**Pull request by sleepylee**
## What happened 👀

Adding more unit test: SurveyRepository > getSurveys()
 
## Insight 📝
Some new techniques: 
- Read test resources (.json) from our test dir.
- Validate all flow of `Store.reset()` - because it relies on a real implementation, so the setup has to be changed with a FakeStorage. Inside there, we can manage to populate any flag we wanted for testing purpose. 
- Not sure why `verify()` cannot be called to check `reset()` was called `1 times` 🤔  it seems like it works only with Mock not Fake object. Anyway, because it's a Fake test class, I modify the data a bit so it's easy to test.
 
## Proof Of Work 📹

Codecov ++

### CODE REVIEW

**Comment by manh-t** [comment]
yeah, it seems be able to `verify` on mock object. Anw, both approaches would work for me 🐎 
image

**Review by manh-t — APPROVED** [review]

**Comment by sleepylee** [comment]
@manh-t but it's weird that I can't use the real `GraphQLCache` instance, I had to mock it, but actually the result is the same 😑  my opinion is that: test as many real objects as possible, only mock when there is a blocker/constructed params.

**Comment by manh-t** [comment]
@sleepylee it's intended behavior. `verify` only work on mock object :harold:
image

File: test/repositories/survey_repository_test.dart [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      ),
+    );
+
+    test('When get Surveys without forceNetworkData, it returns ListSurvey',
+        () async {
+      final result = await testedSurveyRepository.getSurveys("", false);
+
+      expect(injectedFakeStorage.isReset, false);
+      expect(result, isA<List<Survey>>());
+      expect(result.first.title, "Scarlett Bangkok");
```
Comment (luongvo) [inline test/repositories/survey_repository_test.dart:39]: should we assert the list size also?

File: test/repositories/survey_repository_test.dart [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    });
+
+    test(
+        'When get Surveys with forceNetworkData, it reset cache store and returns ListSurvey',
+        () async {
+      final result = await testedSurveyRepository.getSurveys("", true);
+
+      expect(result, isA<List<Survey>>());
+      expect(result.first.title, "Scarlett Bangkok");
```
Comment (luongvo) [inline test/repositories/survey_repository_test.dart:48]: same

File: test/repositories/survey_repository_test.dart (hunk 1 above)
Comment (sleepylee) [reply test/repositories/survey_repository_test.dart:39]: not necessary, we only needs to check if it's parsed correctly, the data size can be changed flexibly based on the test data set, I don't want to tie the logic of the test data to the test assertion of the parser, it's inflexible

**Comment by sleepylee** [comment]
@manh-t even when I mocked the `Store`+ `GraphQLClient` but using a real GraphQLCache() - it won't work...that's what I tried first time

**Comment by manh-t** [comment]
ah, maybe I understand sth here. I try to use `MockStore` this time and it works, @sleepylee 🎉 
image

**Comment by sleepylee** [comment]
@manh-t 🤔 still weird lol 
image

**Review by luongvo — APPROVED** [review]

**Comment by sleepylee** [comment]
@manh-t so if we call like this: 
`verify(mock.real.mock.method())` -> will not work
image



`verify(real.mock.method()` -> works

image

**Comment by manh-t** [comment]
@sleepylee yeah, you're fasttt 🤪 not sure why we can't refer to the real object from mock object 💭

**Comment by sleepylee** [comment]
nah I'd still prefer the current solution then...lol

**Pull request merged**